N/NetCDF/build_tarballs.jl

```diff
@@ -0,0 +1,93 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "NetCDF"
+version = v"4.7.4"
+
+# Collection of sources required to build NetCDF
+sources = [
+    ArchiveSource("https://github.com/Unidata/netcdf-c/archive/v$(version).zip",
+                  "170c9c9020f8909811b06e1034d5ea9288b3d5bd90793e3dd27490191faa7566")
+]
+
+# HDF5.h in /workspace/artifacts/805ccba77cd286c1afc127d1e45aae324b507973/include
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/netcdf-c-*
+
+export CPPFLAGS="-I$prefix/include"
+export LDFLAGS="-L${libdir}"
+
+if [[ ${target} == *-mingw* ]]; then
+   export LDFLAGS="-L${libdir} -lhdf5-0 -lhdf5_hl-0 -lcurl-4"
+
+   if [ ! -f ${WORKSPACE}/destdir/bin/libzlib1.dll ]; then
+       ln -s ${WORKSPACE}/destdir/bin/libz.dll ${WORKSPACE}/destdir/bin/libzlib1.dll
+   fi
+
+   ./configure --prefix=$prefix --build=${MACHTYPE} --host=${target}  --disable-utilities --enable-shared --disable-static
+
+   # linking fails with: "libtool:   error: can't build x86_64-w64-mingw32 shared library unless -no-undefined is specified"
+   # unless -no-undefined is added to LDFLAGS
+   make LDFLAGS="-no-undefined -L$prefix/bin -lhdf5-0 -lhdf5_hl-0 -lcurl-4" -j${nproc}
+
+elif [[ "${target}" == *-apple-* ]]; then
+    # this file is referenced by hdf.h by not installed
+    touch /workspace/destdir/include/features.h
+
+    if [ ! -f ${libdir}/libhdf5.dylib ]; then
+        ln -s ${libdir}/libhdf5.*.dylib ${libdir}/libhdf5.dylib
+    fi
+
+    if [ ! -f ${libdir}/libhdf5_hl.dylib ]; then
+        ln -s ${libdir}/libhdf5_hl.*.dylib ${libdir}/libhdf5_hl.dylib
+    fi
+
+   ./configure --prefix=$prefix --build=${MACHTYPE} --host=${target}  --disable-utilities --enable-shared --disable-static
+   make -j${nproc}
+else
+    # do not exist on Linux(:x86_64)
+    if [ ! -f ${libdir}/libhdf5.so ]; then
+        ln -s ${libdir}/libhdf5.so.* ${libdir}/libhdf5.so
+    fi
+
+    if [ ! -f ${libdir}/libhdf5_hl.so ]; then
+        ln -s ${libdir}/libhdf5_hl.so.* ${libdir}/libhdf5_hl.so
+    fi
+
+   ./configure --prefix=$prefix --build=${MACHTYPE} --host=${target}  --disable-utilities --enable-shared --disable-static
+   make -j${nproc}
+fi
+
+make install
+nc-config --all
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+# Set equal to the supported platforms in HDF5
+platforms = [
+    Linux(:x86_64),
+    Linux(:i686),
+    Linux(:armv7l, libc=:glibc, call_abi=:eabihf),
```

Comment:
I had a look at the failure on this platform, I'd suggest to just remove this
```suggestion
```
HDF5 for this platform comes from a custom native build provided by a user, which however is incompatible with our build environment.  This is the error message in the build environment:
```
configure:17887: cc -o conftest    conftest.c -lhdf5  -lm -ldl -lz -lcurl  >&5
/opt/arm-linux-gnueabihf/bin/../arm-linux-gnueabihf/sys-root/usr/local/lib/libhdf5.so: undefined reference to `fcntl64@GLIBC_2.28'
/opt/arm-linux-gnueabihf/bin/../arm-linux-gnueabihf/sys-root/usr/local/lib/libhdf5.so: undefined reference to `powf@GLIBC_2.27'
collect2: error: ld returned 1 exit status
```
but, as shown in [`RootFS.md`](https://github.com/JuliaPackaging/Yggdrasil/blob/d1e5159beef7fcf8c631e893f62925ca5bd54bec/RootFS.md#glibc-versions), `armv7l` has glibc v2.19.  I don't think we can do anything here from within the build environment.

The long term solution would be to finally have a native build of HDF5

Reply:
OK,thank you. I just updated my PR removing armv7l (but I feel sorry for the Raspberry Pi users ...)

Reply:
> but I feel sorry for the Raspberry Pi users ...

At least it's not a regression :wink: